File: game_server/ServerProtocol/src/NPEnum/ENPShareItemType.java
package NPEnum;

/*********
 * 分享物品类型
 **/
public enum ENPShareItemType {
	NONE, //0 ==== 
	BOX, //1 ==== 宝箱
	;
public static final ENPShareItemType[]  ENPShareItemType_Values = ENPShareItemType.values();
public static final int ENPShareItemType_Length = ENPShareItemType_Values.length;
public static ENPShareItemType ENPShareItemType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ENPShareItemType_Length){ return null; }
	return ENPShareItemType_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/NPEnum/ENPShareItemType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace NPEnum
{

/// <summary>
/// 分享物品类型
/// </summary>
public enum ENPShareItemType {
	NONE, //0 ==== 
	[InspectorName("BOX - idx[1] - 宝箱")]
	BOX, //1 ==== 宝箱
}

public class ENPShareItemTypeComparer : IEqualityComparer<ENPShareItemType>{
	public bool Equals(ENPShareItemType x, ENPShareItemType y) { return x == y; }
	public int GetHashCode(ENPShareItemType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 2;
}
}

